source/common/http/conn_manager_impl.cc

```diff
@@ -1531,18 +1531,30 @@ void ConnectionManagerImpl::ActiveStream::onBelowWriteBufferLowWatermark() {
 }
 
 Tracing::OperationName ConnectionManagerImpl::ActiveStream::operationName() const {
+  ASSERT(
+      connection_manager_.config_.tracingConfig(),
+      "tracingConfig() must be called first and verified it's not null before calling this method");
   return connection_manager_.config_.tracingConfig()->operation_name_;
 }
 
 const Tracing::CustomTagMap* ConnectionManagerImpl::ActiveStream::customTags() const {
+  ASSERT(
+      connection_manager_.config_.tracingConfig(),
+      "tracingConfig() must be called first and verified it's not null before calling this method");
   return tracing_custom_tags_.get();
 }
 
 bool ConnectionManagerImpl::ActiveStream::verbose() const {
+  ASSERT(
```

Comment:
this is called from https://github.com/envoyproxy/envoy/blob/master/source/common/tracing/http_tracer_impl.cc#L252, the consumer of that method needs a nullptr check.